web/src/message_view.ts

```diff
@@ -295,6 +295,11 @@ export function try_rendering_locally_for_same_narrow(
     filter: Filter,
     opts: ShowMessageViewOpts,
 ): boolean {
+    if (!narrow_state.is_message_feed_visible()) {
+        // This function only works when the message feed is visible.
+        return false;
+    }
```

Comment:
This probably works, but it feels to me that this is probably a sign that we're doing something wrong with abstractions.